File: Anushka-Pachauri/SlidingWindow/MaxConsecutiveOnesIII.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int start = 0;
        int end = 0;
        int zeroCount = 0;
        int maxLength = 0;
        while (end < nums.length) {
            if (nums[end] == 0) {
                zeroCount++;
            }
            while (zeroCount > k) {
                if (nums[start] == 0) {
                    zeroCount--;
                }
                start++;
            }
            maxLength = Math.max(maxLength, end - start + 1);
            end++;
        }
        return maxLength;
    }
}

File: Ashutosh-Singh/SlidingWindow/MaxConsecutive-Ones-III.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int n = nums.size();
        int zeroCnt = 0;
        int maxLen = 0;
        int left = 0;

        for (int right = 0; right < n; ++right) {
            if (nums[right] == 0) {
                zeroCnt++;
            }
            while (zeroCnt > k) {
                if (nums[left] == 0) {
                    zeroCnt--;
                }
                left++;
            }
            maxLen = max(maxLen, right - left + 1);
        }

        return maxLen;
    }
};

int main() {
    int n, k;
    cin >> n >> k;

    vector<int> nums(n);
    for (int i = 0; i < n; ++i) {
        cin >> nums[i];
    }

    Solution solution;
    cout << solution.longestOnes(nums, k) << endl;

    return 0;
}
